rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java

```diff
@@ -1538,4 +1538,92 @@ public List<ReleaseLink> serveLinkedReleasesInDependencyNetworkByIndexPath(Strin
+        ThriftClients thriftClients = new ThriftClients();
+        ProjectService.Iface projectClient = getThriftProjectClient();
+        LicenseInfoService.Iface licenseInfoClient = thriftClients.makeLicenseInfoClient();
+        ComponentService.Iface componentClient = thriftClients.makeComponentClient();
+        JSONObject jsonResult = new JSONObject();
+
+        try {
+            Project project = projectClient.getProjectById(projectId, sw360User);
+            if (project == null) {
+                throw new IllegalArgumentException("Project with ID " + projectId + " not found.");
+            }
+
+            Set<String> releaseIds = CommonUtils.getNullToEmptyKeyset(project.getReleaseIdToUsage());
+            List<Release> releasesWithSingleCLI = new ArrayList<>();
+            List<Release> releasesWithMultipleCLI = new ArrayList<>();
+
+            for (String releaseId : releaseIds) {
+                Release release = componentClient.getReleaseById(releaseId, sw360User);
+                if (release == null) {
+                    throw new IllegalArgumentException("Release with ID " + releaseId + " not found.");
+                }
+
+                List<Attachment> approvedCliAttachments = SW360Utils.getApprovedClxAttachmentForRelease(release);
+                if (approvedCliAttachments.isEmpty()) {
+                    approvedCliAttachments = SW360Utils.getClxAttachmentForRelease(release);
+                }
+
+                if (approvedCliAttachments.size() == 1) {
+                    processSingleAttachment(approvedCliAttachments.get(0), release, licenseInfoClient, sw360User);
+                    releasesWithSingleCLI.add(release);
+                } else {
+                    if (approvedCliAttachments.size() > 1) {
+                        releasesWithMultipleCLI.add(release);
+                    }
+                    jsonResult.put(SW360Constants.STATUS, SW360Constants.FAILURE);
+                }
+
+                componentClient.updateRelease(release, sw360User);
+            }
+
+            jsonResult.put("releasesWithSingleCLI", releasesWithSingleCLI);
+            jsonResult.put("releasesWithMultipleCLI", releasesWithMultipleCLI);
+
+        } catch (Exception e) {
+            throw new RuntimeException("Error processing linked releases for project: " + projectId, e);
+        }
+
+        return jsonResult;
+    }
+
+    private void processSingleAttachment(Attachment attachment, Release release,
+            LicenseInfoService.Iface licenseInfoClient, User sw360User) throws TTransportException, TException {
+        String attachmentName = attachment.getFilename();
+        Set<String> mainLicenses = release.getMainLicenseIds() != null ? release.getMainLicenseIds() : new HashSet<>();
+        Set<String> otherLicenses = release.getOtherLicenseIds() != null ? release.getOtherLicenseIds()
+                : new HashSet<>();
```

Comment:
In accordance with #2827, the main licenses and other licenses of the release should be overwritten.

Thus the `mainLicenses` and `otherLicenses` should always be created as fresh `HashSet`.